core/modules/user/user.admin.inc

```diff
@@ -508,16 +524,11 @@ function user_admin_roles($form, $form_state) {
     '#collapsed' => TRUE,
     '#weight' => -20,
   );
-  // Do not allow the anonymous or authenticated user roles as administrator.
-  $roles = user_roles();
-  unset($roles[BACKDROP_ANONYMOUS_ROLE]);
-  unset($roles[BACKDROP_AUTHENTICATED_ROLE]);
-  $roles[0] = t('disabled');
   $form['admin_role']['user_admin_role'] = array(
     '#type' => 'select',
     '#title' => t('The role that will be automatically assigned new permissions whenever a module is enabled:'),
-    '#default_value' => $system_config->get('user_admin_role'),
-    '#options' => $roles,
+    '#options' => $options,
+    '#default_value' => $system_config->get('user_admin_role') == NULL ? $options[0] : $system_config->get('user_admin_role'),
```

Comment:
...there's a slight (pre-existing) bug here, in that if no role is set as admin, the first available role in the select is selected as the default value.